File: HtOMSBrief/BriefModels.swift
import Foundation

struct ChannelShare: Identifiable, Sendable, Equatable {
    let name: String
    let count: Int
    let percentage: Int

    var id: String { name }
}

struct SlicePortion: Identifiable, Sendable, Equatable {
    let name: String
    let value: Double

    var id: String { name }
}

struct DailyPoint: Identifiable, Sendable, Equatable {
    let label: String
    let reference: Int?
    let result: Int?

    var id: String { label }
}

struct MonthlyPoint: Identifiable, Sendable, Equatable {
    let date: Date
    let count: Int

    var id: Date { date }
}

struct TodaySales: Sendable, Equatable {
    let day: Int
    let level: String
    let amount: Int
}

struct ServerStatus: Identifiable, Sendable, Equatable {
    let name: String
    let isOperational: Bool

    var id: String { name }
}

enum DeliveryStatus: String, CaseIterable, Sendable {
    case preparing = "준비"
    case accepted = "인수"
    case moving = "이동"
    case departing = "출발"
    case completed = "완료"
    case invoiceError = "송장오류"
    case undelivered = "미배달"
    case unavailable = "배송불가"
}

struct DeliveryStatusCount: Identifiable, Sendable, Equatable {
    let status: DeliveryStatus
    let count: Int

    var id: DeliveryStatus { status }
}

struct BriefOverview: Sendable, Equatable {
    let todaySales: TodaySales
    let channels: [ChannelShare]
    let monthLabel: String
    let monthProgress: Int
    let monthTotal: Int
    let monthAverage: Int
    let levelMix: [SlicePortion]
    let categoryMix: [SlicePortion]
    let refreshedAt: String
    let serverStatuses: [ServerStatus]
}

struct DeliverySummary: Sendable, Equatable {
    let dateRange: String
    let statuses: [DeliveryStatusCount]

    var total: Int { statuses.reduce(0) { $0 + $1.count } }
}

struct BriefSnapshot: Sendable, Equatable {
    let overview: BriefOverview
    let dayTrend: [DailyPoint]
    let monthTrend: [MonthlyPoint]
    let monthAverage: Int
    let deliverySummary: DeliverySummary
    var widgetSnapshot: DashboardWidgetSnapshot? = nil
}

File: app/src/main/java/com/htoms/brief/model/BriefModels.kt
package com.htoms.brief.model

import kotlinx.serialization.Serializable

@Serializable
data class ChannelShare(
    val name: String,
    val count: Int,
    val percentage: Int
) {
    val id: String get() = name
}

@Serializable
data class SlicePortion(
    val name: String,
    val value: Double
) {
    val id: String get() = name
}

@Serializable
data class DailyPoint(
    val label: String,
    val reference: Int?,
    val result: Int?
) {
    val id: String get() = label
}

@Serializable
data class MonthlyPoint(
    val date: Long,
    val count: Int
) {
    val id: Long get() = date
}

@Serializable
data class TodaySales(
    val day: Int,
    val level: String,
    val amount: Int
)

@Serializable
data class ServerStatus(
    val name: String,
    val isOperational: Boolean
) {
    val id: String get() = name
}

@Serializable
enum class DeliveryStatus(val label: String) {
    PREPARING("준비"),
    ACCEPTED("인수"),
    MOVING("이동"),
    DEPARTING("출발"),
    COMPLETED("완료"),
    INVOICE_ERROR("송장오류"),
    UNDELIVERED("미배달"),
    UNAVAILABLE("배송불가")
}

@Serializable
data class DeliveryStatusCount(
    val status: DeliveryStatus,
    val count: Int
) {
    val id: DeliveryStatus get() = status
}

@Serializable
data class BriefOverview(
    val todaySales: TodaySales,
    val channels: List<ChannelShare>,
    val monthLabel: String,
    val monthProgress: Int,
    val monthTotal: Int,
    val monthAverage: Int,
    val levelMix: List<SlicePortion>,
    val categoryMix: List<SlicePortion>,
    val refreshedAt: String,
    val serverStatuses: List<ServerStatus>
)

@Serializable
data class DeliverySummary(
    val dateRange: String,
    val statuses: List<DeliveryStatusCount>
) {
    val total: Int get() = statuses.sumOf { it.count }
}

@Serializable
data class BriefSnapshot(
    val overview: BriefOverview,
    val dayTrend: List<DailyPoint>,
    val monthTrend: List<MonthlyPoint>,
    val monthAverage: Int,
    val deliverySummary: DeliverySummary,
    val widgetSnapshot: DashboardWidgetSnapshot? = null
)
